Added getWidth function to correctly calculate parent's width and account for padding in parent.

andycarlberg opened the pull request:
Base fork version failed to account for padding.  offsetWidth doesn't account for padding on it's own.  There was also a magic number '6' (perhaps to account for padding?).  Correctly calculated available width of parent now.

skoberlink committed:
Added getWidth function to correctly calculate parent's width.

Previous version failed to account for padding.  offsetWidth doesn't account for padding on it's own.  There was also a magic number '6' (perhaps to account for padding?).  Correctly calculated available width of parent now.

skoberlink commented:
Sorry, this is phrased poorly.  Using offsetWidth tries to make the target element fit the ratio of the parent including it's padding.  We want it to fit within the width of the parent without padding.  The new function calculates the appropriate width value without the padding.  It is also possible the function should account for border, however, I did not test this.

skoberlink committed:
Added support for IE <9

skoberlink committed:
Account for an invalid ratio

If the element in question does not currently have a width (possibly due
to not being shown at the time), ratio is NaN.  This will correct a NaN
ratio to 0.

skoberlink committed:
Switched timeout to interval

Changed timeout to interval so the resizer is constantly checking.  This will let the resizer update if the innerHTML changes.  Could not find an event to tie to so this is the best fix I could come up with.

patrickmarabeas commented:
Tanks, I'll have to give this a look through.

Unsure why `$interval` is required - `scope.$watch` on the next line should achieve exactly that.

skoberlink committed:
Changed $interval back to $timeout

skoberlink commented:
Ah, you're right.  I don't know what I was thinking, must have been asleep at the wheel.  I changed it back and used mg-model.  It works correctly for me now.

skoberlink committed:
IE <9 fix for getWidth Function

had the wrong style name in the getWidth function.  That's what I get for using an emulator...

skoberlink committed:
Undo remnants of switching timeout to interval

Still had some extra code from switching timeout to interval that was causing problems.  Really should have read the documentation better...

SuricateCan commented:
Is this PR ever going to be merged?

helarqjsc commented:
Is the project completely abandoned at this point?

patrickmarabeas commented:
No, life just getting in the way.

Would be helpful if bugs had demos so fixes can actually be tested.

patrickmarabeas commented:
Please confirm whether v4.2.2 works/doesn't work; code base has changed a fair amount since this PR - simply applying the `getWidth` function doesn't solve the magic number `6` in context of demo.

Please create a codepen demonstrating issue and update branch with master.

patrickmarabeas commented:
Additionally if you can confirm #53 fixes the issue, I'll merge that in as it doesn't look to break existing demos (v4.2.3)

andycarlberg commented:
I know this is years too late but I'm reviewing some old repos and found this PR again. I believe #53 fixes the issue and is obviously a more up to date PR. I'm happy closing this one.
